core/src/test/java/org/infinispan/executors/LimitedExecutorTest.java

```diff
@@ -57,11 +57,13 @@ public void testConcurrencyLimit() throws Exception {
       limitedExecutor.execute(() -> cf2.complete("bla"));
 
       Thread.sleep(10);
+      assertFalse(cf1.isDone());
       assertFalse(cf2.isDone());
 
       latch.countDown();
       assertEquals("bla", cf1.get(10, SECONDS));
       assertEquals("bla", cf2.get(10, SECONDS));
+      eventuallyEquals(0, executor::getActiveCount);
```

Comment:
I would still add a check into the second task verifying that the first task has finished; otherwise you could replace LimitedExecutor with some executor that starts executing the second task a bit later (than the 10 ms) and it would still succeed.